File: Day3/Part2.py
total = 0

first, second, third = [], [], []
with open("C:\\Users\\User\\Downloads\\advent.txt") as file:
    for line in file:
        items = line.strip()

        if not first:
            first = items
            continue
        elif not second:
            second = items
            continue
        else:
            third = items

        value = 0
        for item in first:
            if item in second and item in third:
                value = ord(item)
                break

        first, second, third = [], [], []

        if value > 97:
            value -= 96
        else:
            value -= 38

        total += value

print(total)

File: 2022/Day1/Part2.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace AdventOfCode.Day1
{
    class Part2
    {
        static void Temp()
        {
            string[] lines = System.IO.File.ReadAllLines(@"C:\Users\User\Downloads\advent.txt");

            int currCalories = 0;
            List<int> mostCalories = new List<int>() { 0, 0, 0 };

            foreach (string line in lines)
            {
                if (line == "")
                {
                    if (currCalories > mostCalories[0])
                    {
                        mostCalories.RemoveAt(0);

                        for (int index = 0; index < mostCalories.Count; index++)
                        {
                            if (currCalories < mostCalories[index])
                            {
                                mostCalories.Insert(index, currCalories);
                                break;
                            }
                        }

                        if (mostCalories.Count != 3) mostCalories.Add(currCalories);
                    }

                    currCalories = 0;
                    continue;
                }

                currCalories += Int32.Parse(line);
            }

            if (currCalories > mostCalories[0])
            {
                mostCalories.RemoveAt(0);
                mostCalories.Add(currCalories);
            }

            Console.WriteLine($"Calories: {mostCalories.Sum()}");
        }
    }
}
